cmd/nerdctl/pull.go

```diff
@@ -108,5 +128,78 @@ func pullAction(cmd *cobra.Command, args []string) error {
 
 	_, err = imgutil.EnsureImage(ctx, client, cmd.OutOrStdout(), cmd.ErrOrStderr(), snapshotter, args[0],
 		"always", insecure, ocispecPlatforms, unpack, quiet)
-	return err
+
+	if err != nil {
+		return err
+	}
+
+	switch verifier {
+	case "cosign":
+		keyRef, err := cmd.Flags().GetString("cosign-key")
+		if err != nil {
+			return err
+		}
+
+		if err := verifyCosign(ctx, rawRef, keyRef); err != nil {
+			return err
+		}
+	case "none":
+		logrus.Debugf("verification process skipped")
+	default:
+		return fmt.Errorf("no verifier found: %s", verifier)
+	}
+	return nil
+}
+
```

Comment:
The race https://github.com/containerd/nerdctl/pull/556#discussion_r759121329 seems introduced again in the latest revision.

`imgutil.EnsureImage` should be called with the verified digest value when `verifier == "cosign"`